app/sign.js

```diff
@@ -103,7 +109,12 @@ const handleSignUtxo = function(recoveryRequest, key, skipConfirm) {
   // force override network as we use btc mainnet xpubs for all utxo coins
   backupKeyNode.keyPair.network = network;
 
+  transaction.ins.forEach(function (input, i) {
+    transaction.ins[i].value = recoveryRequest.inputs[i].amount;
+  })
+
   const txBuilder = utxoLib.TransactionBuilder.fromTransaction(transaction, network);
+  const isBCH = recoveryRequest.coin === 'bch' || recoveryRequest.coin === 'tbch';
```

Comment:
Can we do two things here:

1. Move this const definition closer to where it is being used (right about line 132 -- next to the conditional). In fact, you may be able to get away with doing this all in one line like `if (recoveryRequest.coin === 'bch' || recoveryRequest.coin === 'tbch'){ ... do custom bch stuff };` without having to declare a constant.
1. Add an inline comment around why this conditional is necessary for BCH in the first place.